04/app.js

```diff
@@ -1,89 +1,105 @@
-import React from 'react';
-import { createRoot } from 'react-dom/client';
+import React from "react";
+import { createRoot } from "react-dom/client";
 
-const root = createRoot(document.querySelector('#root'));
+const root = createRoot(document.querySelector("#root"));
 
 class App extends React.Component {
-    state = { 
-        firstName: '',
-        lastName: '',
-        searchQuery: '',
-        users: ['Jan Kowalski', 'Michał Nowak'],
-    }
+  state = {
+    firstName: "",
+    lastName: "",
+    searchQuery: "",
+    users: ["Jan Kowalski", "Michał Nowak"],
+  };
 
-    renderUsersList() {
-        const {users} = this.state;
-        return users.map(name => {
-            return (
-                <li onClick={ this.clickHandler }>
-                    { name }
-                </li>
-            );
-        });
-    }
+  generateKey = (pre) => {
+    return `${pre}_${new Date().getTime()}`;
+  };
 
-    clickHandler = e => {
-        const {innerText: userName} = e.target;
-        this.removeUser(userName);
-    }
+  renderUsersList() {
+    const { users, searchQuery } = this.state;
 
-    inputChange = e => {
-        const {name, value} = e.target;
-        this.setState({
-            [name]: value,
-        });
+    let currUsers = [];
+    if (searchQuery !== "") {
+      currUsers = users.filter((user) =>
+        user.toLowerCase().includes(searchQuery.toLowerCase())
+      );
+    } else {
+      currUsers = users;
     }
 
-    render() {
-        const { firstName, lastName } = this.state;
-        return (
-            <section onSubmit={ this.submitHandler }>
-                <form>
-                    <input name="firstName"
-                        value={ firstName }
-                        onChange={ this.inputChange }
-                    />
-                    <input name="lastName"
-                        value={ lastName }
-                        onChange={ this.inputChange }
-                    />
-                    <input type="submit"/>
-                </form>
-                <ul>{ this.renderUsersList() }</ul>
-            </section>
-        );
-    }
+    return currUsers.map((name) => {
+      return (
+        <li key={this.generateKey(name)} onClick={this.clickHandler}>
+          {name}
+        </li>
+      );
+    });
+  }
 
-    submitHandler = e => {
-        e.preventDefault();
+  clickHandler = (e) => {
+    const { innerText: userName } = e.target;
+    this.removeUser(userName);
+  };
 
-        const { firstName, lastName } = this.state;
-        if(firstName && lastName) {
-            this.addUser(`${firstName} ${lastName}`);
-            this.setState({
-                firstName: '',
-                lastName: '',
-            });
-        } else {
-            // tutaj komunikat dla użytkownika
-        }
-    }
+  inputChange = (e) => {
+    const { name, value } = e.target;
+    this.setState({
+      [name]: value,
```

Comment:
Czyli tutaj np. zrobił tak: ` [name]: {user: value, id: this.generateKey(name)},`